vignettes/duckdb_arrow.Rmd

```diff
@@ -0,0 +1,261 @@
+---
+title: "Using DuckDB and Arrow with piggyback"
+output: rmarkdown::html_vignette
+author: "Tan Ho, Carl Boettiger"
+date: "2023-12-26"
+vignette: >
+  %\VignetteIndexEntry{Using DuckDB and Arrow with piggyback}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+```
+```{r}
+library(piggyback)
+```
+## Data Too Big To Fit In Memory
```

Comment:
All of this material is new. I spent time digging and realized that while you can read one file from URL into arrow, you can't read a vector of URLs yet (tracked in apache/arrow#18980). 

I started writing out using pb_download + arrow filesystem-dataset, but realized that's not really doing anything special so just wrapped it all into an html comment for future usage.

Reply:
:bangbang:   I think the arrow code that reads from a single url is only for the in-memory reading mechanisms though (`read_parquet` / `read_csv`) right?   

By the way -- note that we can read a vector of URLs in now from https with a single line of code using my very simple duckdb wrapper:

```r
base <- paste0("https://github.com/duckdb/duckdb/raw/main/",
               "data/parquet-testing/hive-partitioning/union_by_name/")
f1 <- paste0(base, "x=1/f1.parquet")
f2 <- paste0(base, "x=1/f2.parquet")
urls <- c(f1,f2)

duckdbfs::open_dataset(urls)
```

https://github.com/cboettig/duckdbfs?tab=readme-ov-file#quickstart

Reply:
> ‼️ I think the arrow code that reads from a single url is only for the in-memory reading mechanisms though (read_parquet / read_csv) right?

For now :tm: I believe it's on the roadmap and will be resolved eventually?
In any case I think I might rename the vignette to something like cloud_native so that we can come back to arrow in the future

Reply:
duckdbfs looks great! I might append it as a section below as an option to streamline the duckdb. Do you know of a way to do hive-style partitioning in the filenames of piggyback? (my guess is that it's not possible because no path separators in github release asset names?)

Reply:
ah, good question.  yup, pretty sure it's not possible for that reason :/

Reply:
Added duckdbfs mention and went ahead with rename to cloud_native
https://github.com/ropensci/piggyback/pull/112/files#diff-b1603f8848f47d86de1bd9366dad7062e8a8de6efa9ff125f214a2f99afb513d